gobblin-metrics-libs/gobblin-metrics-base/src/main/avro/DagActionStoreChangeEvent.avsc

```diff
@@ -0,0 +1,47 @@
+{
+  "type" : "record",
+  "name" : "DagActionStoreChangeEvent",
+  "namespace" : "org.apache.gobblin.service.monitoring",
+  "doc" : "Contains information to identify a change that occurred in the DagActionStore which contains pending kill or resume requests",
+  "fields" : [ {
+    "name" : "key",
+    "type" : "string",
+    "doc" : "Primary key for the DagActionStore (flowGroup, flowName, flowExecutionId)",
```

Comment:
What will be the key here, a concat string of these there values?

Reply:
> Thanks for this PR, one main comment is can we merge the dagActionStoreMonitor with the SpecStoreChangeMonitor, as they essentially share the same logic, just for different event it needs to take different action. We can even modify the specStoreChangeEvent to be more general that it can contain two types of data? So that we can leverage most of the code and only have one consumer to consume all the change stream events. WDYT?

responding to this comment first. I also see lots of similarities in both other than the action we take based on processing each event. However the classes are not repeating much else other than the structure. Here are a couple reasons for keeping separate: 

1) `HighLevelConsumer` only listens to one topic and each database has own Kafka topic. Having multiple topics in the derived class will break this model. 
-> Instead I can make some of the repeated logic into re-usable methods. 

2) It will be a bit hacky to try to combine the two events together to handle these two use cases, if we add more stores in the future we will need to combine more properties for each event. 
-> It could be better to instead of a base event that we reuse (contains common properties like opCode, timestamp, key) and then extend additional properties as needed. 

What do you think?

Reply:
The key should be concatenation of these values but I do not know if there's any delimiter. Need to verify with the Brooklin and/or Mysql team. They also did not seem to know.

Reply:
Yeah, have a base event and abstract common logic seems reasonable to me as well

Reply:
Updated after abstracting out some common logic. The insert/update/delete handling is hard to abstract out as we have very minute differences for each. I also created a base event.